internal/services/recoveryservices/site_recovery_replication_recovery_plan_resource.go

```diff
@@ -517,14 +517,26 @@ func (r SiteRecoveryReplicationRecoveryPlanResource) Update() sdk.ResourceFunc {
 				return fmt.Errorf("parse Site reocvery replication plan id: %+v", err)
 			}
 
-			recoveryPlanGroup, err := expandRecoveryGroup(model.RecoveryGroup)
 			if err != nil {
```

Comment:
Move this below the new code and remove the duplicate extra error return code in the if-else structure.